app/assets/javascripts/dradis/plugins/csv/upload.js

```diff
@@ -1,12 +1,38 @@
 window.addEventListener('job-done', function(e){
-  var jobId = e.detail.job_id,
-      uploader = document.getElementById('uploader');
+  if ($('body.upload.index').length) {
+    var jobId = e.detail.job_id,
+        uploader = document.getElementById('uploader');
 
-  if (uploader.value === 'Dradis::Plugins::CSV') {
-    var path = window.location.pathname;
-    var project_path = path.split('/').slice(0, -1).join('/');
+    if (uploader.value === 'Dradis::Plugins::CSV') {
+      var path = window.location.pathname;
+      var project_path = path.split('/').slice(0, -1).join('/');
 
-    var redirectPath  = project_path + '/csv/upload/new?job_id=' + jobId;
-    Turbolinks.visit(redirectPath);
+      var redirectPath  = project_path + '/csv/upload/new?job_id=' + jobId;
+      Turbolinks.visit(redirectPath);
+    }
+  }
+});
+
+document.addEventListener('turbolinks:load', function() {
+  if ($('body.upload.new').length) {
```

Comment:
this seems not specific enough for this engine, I wouldn't be surprised if other engines end up with a very similar body class collection, no? /cc @MattBudz 

we're probably better off w/ a specific CSV data-behavior on some container element.

Reply:
Agreed, I already have a task for immediately after this cycle to update the Tylium layout. We are using the `controller_name` as a class rather than the `controller_path` like we are in mintcreek. This will touch a lot of scripts and stylesheets so I didn't want to pollute this cycle's work with this change.